R/robin_glm.R

```diff
@@ -53,21 +47,55 @@ robin_glm <- function(
       "using a linear model without treatment-covariate interactions; see the 2023 FDA guidance."
     )
   }
+
+  if (identical(family$family, "Negative Binomial(NA)")) {
+    fit <- MASS::glm.nb(formula, data = data, ...)
+  } else {
+    fit <- glm(formula, family = family, data = data, ...)
+  }
+  pc <- eval(bquote(predict_counterfactual(fit, treatment, data, vcov = .(substitute(vcov)), vcov_args = vcov_args)))
   if (missing(pair)) {
     pair <- pairwise(names(pc$estimate))
   }
-  if (identical(contrast, "difference")) {
-    difference(pc, pair = pair)
-  } else if (identical(contrast, "risk_ratio")) {
-    risk_ratio(pc, pair = pair)
-  } else if (identical(contrast, "odds_ratio")) {
-    odds_ratio(pc, pair = pair)
+
+  if (
+    test_string(contrast) &&
+      contrast %in% c("difference", "risk_ratio", "odds_ratio", "log_risk_ratio", "log_odds_ratio")
```

Comment:
Just an idea: how about putting all these options in the function signature and then use `match.arg()` inside?

Reply:
updated